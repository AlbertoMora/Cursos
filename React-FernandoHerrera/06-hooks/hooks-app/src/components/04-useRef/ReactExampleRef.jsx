import React, { useState } from 'react'

import MultipleCustomHooks from '../03-examples/MultipleCustomHooks';

import '../02-useEffect/effects.css';

const ReactExampleRef = () => {

    const [show, setShow] = useState(false);

    const handleToggle = () => {
        setShow(!show);
    }

    return (
        <>
            <h1>RealExapleRef</h1>
            <hr />
            <button onClick={handleToggle} className="btn btn-primary mt-5">
               Show/Hide
            </button>
            
           { show && <MultipleCustomHooks /> }

           <hr />

           
        </>
    )
}

export default ReactExampleRef
