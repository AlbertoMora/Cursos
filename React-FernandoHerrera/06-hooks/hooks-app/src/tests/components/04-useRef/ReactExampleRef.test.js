import React from "react";
import { shallow } from "enzyme";

import ReactExampleRef from "../../../components/04-useRef/ReactExampleRef";

describe("Test on <ReactExampleRef />", () => {
    const wrapper = shallow(<ReactExampleRef />);

    test("should render correctly", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test("should show component <MultipleCustomHooks />", () => {
        wrapper.find("button").at(0).simulate("click");

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
    });
});
