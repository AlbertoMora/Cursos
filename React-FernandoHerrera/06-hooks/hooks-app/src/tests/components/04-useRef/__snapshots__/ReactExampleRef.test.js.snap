// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <ReactExampleRef /> should render correctly 1`] = `
<Fragment>
  <h1>
    RealExapleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
  <hr />
</Fragment>
`;

exports[`Test on <ReactExampleRef /> should show component <MultipleCustomHooks /> 1`] = `
<Fragment>
  <h1>
    RealExapleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
  <MultipleCustomHooks />
  <hr />
</Fragment>
`;
